### DISCUSSION TITLE
Regarding Use of OpenFast

### DISCUSSION THREAD

**Discussion by KanwarOsama** [OpenFAST/openfast#1747] [Q&A]
Hi,
 I am a civil engineer, trying to work with the Structural health monitoring of wind turbines. During my research, I found NREL OpenFast where I could do the structural analysis of the wind turbines to somehow find the strains, vibrations, deflections of the blades, tower etc. I would also be interested in finding the edgewise and the flapwise bending moments, shear forces and normal forces for the blades and the tower.

My Questions are;

> Would OpenFast be a reasonable option to find such e.g. strains, modes of vibrations, stresses present in the towers and foundations etc. If yes, what are the important modules which i should be focussing on. Or should I be using FEM analysis through abaqus etc. 

> I also understood that ElastoDyn is kind of more important when tower is specifically considered and for that the distributed properties are required as an input. But for a certain tower, are there some formulas that are available and can be used to find these distributed properties. Also, as for the baldes we need the stuctural properties as an input, any sources where i could find them or calculate myself based on some formulas. 

> In order to learn Openfast in general, what are the most important online sources?

Please let me know of your opinions. Also, please guide me if there are some other sources where i can do the static and the dynamic analysis of the wind turbines.  Your opinion is really appreiated.

Regards, 
Kanwar

**Comment by jjonkman** [comment]
Der @KanwarOsama,

Just a few comments:

- OpenFAST provides state-of-the-art coupled aero-hydro-servo-elastic simulation of land-based and offshore fixed-bottom and floating wind turbines with the ability to run large numbers of nonlinear time-domain simulations to enable loads analysis for predicting wind system dynamic response, including ultimate and fatigue loads.
- OpenFAST makes use of beam models of the blades and tower, so, while OpenFAST can compute sectional loads (bending moments, torque, shear forces, axial force) and deflections (displacement/orientation of the cross section), it does not directly calculate stress or strain.  For simple circular cross sections composed of isotropic materials often used in towers, the sectional loads can be related to stress/strain in an analytical post-processing step, but for complicated airfoil cross sections composed of composite materials often used in blades, the conversion from sectional loads to stress/strain requires sophisticated FEA post-processing.
- Likewise, the beam properties (mass per unit length, bending stiffness) are typically straightforward to define (analytically) for circular cross sections composed of isotropic materials often used in towers, but are more difficult to calculate (requiring a sectional analysis preprocessing step) for complicated airfoil cross sections composed of composite materials often used in blades.  See documentation for the beam properties of towers and blades of ElastoDyn in the old FAST v6 User's Guide for more information: https://openfast.readthedocs.io/en/dev/_downloads/d8bd014121d6505cb25cf49bee5eaa80/Old_FAST6_UsersGuide.pdf.
- Do you have a specific wind turbine (that you know the properties of) in mind, or do you intend to use on of our available reference models, such as the popular NREL 5-MW baseline wind turbine or the IEA Wind 15-MW reference wind turbine?
- For land-based wind turbines, important modules of OpenFAST include ElastoDyn (for structural dynamics), AeroDyn (for rotor aerodynamics), InflowWind (for wind inflow), and ServoDyn (for controls).  For blades with significant deflection, built-in curvature, or composite couplings, BeamDyn can be used in place of ElastoDyn to model the blade structural dynamics.
- Regarding learning how to use OpenFAST, see similar questions asked and answered: https://github.com/OpenFAST/openfast/discussions/1339, https://github.com/OpenFAST/openfast/issues/1297, https://github.com/OpenFAST/openfast/issues/626, and our on forum: https://forums.nrel.gov/.

Best regards,
